### ISSUE TITLE
[CMIP6 CMORization & ESGF-publication]  NorESM2-LM - esm-piControl

### ISSUE THREAD

**Issue by tjiputra** [NorESMhub/noresm2cmor#47]
### Mandatory information:

**Full path to the case(s) of the experiment on NIRD**
/projects/NS2345K/noresm/cases/N1850_f19_tn14_20190730esm

**experiment_id**
esm-piControl

**CASENAME(s) and years to be CMORized**
N1850_f19_tn14_20190730esm, 1851-2100

### Optional information
(additional information, if the experiment is branched/hybrid restart from previous parent experiment; you may find more information relevant to the experiment_id here: https://github.com/NorwegianClimateCentre/noresm2cmor/blob/master/tables/CMIP6_CV.json)

**parent_experiment_id**
<strike>piControl</strike>
'esm-piControl-spinup'

**parent_experiment_rip**
e.g., r1i1p1

**parent_time_units**
'days since 1851-01-01'

**branch_method**
'Hybrid-restart from year 1851-01-01 of piControl',

**other information**
(provide other information that might be useful)
Prognostic atmospheric CO2 simulation

**Comment by YanchunHe** [comment]
parent_experiment_id
should be  'esm-piControl-spinup'
according to CMIP6_CV.json
But I guess piControl is actually used as parent id.

I set 'esm-piControl-spinup' to be consistent with CMOR requirement, only to avoid repeated warning reports.

**Comment by YanchunHe** [comment]
@tjiputra  please check

time_axis in some daily micom output may be wrong:

_ncrcat -v time N1850_f19_tn14_20190730esm.micom.hd.1861-*.nc /scratch/yanchun/time_1861.nc
ncrcat: WARNING Intra-file non-monotonicity. Record coordinate "time" does not monotonically increase between (input file N1850_f19_tn14_20190730esm.micom.hd.1861-07b.nc record indices: 0, 1) (output file /scratch/yanchun/time_1861.nc record indices 181, 182) record coordinate values 40357.500000, 0.000000_

$ ncdump -v time -t /scratch/yanchun/time_1861.nc
...
    "1861-06-18 12", "1861-06-19 12", "1861-06-20 12", "1861-06-21 12",
    "1861-06-22 12", "1861-06-23 12", "1861-06-24 12", "1861-06-25 12",
    "1861-06-26 12", "1861-06-27 12", "1861-06-28 12", "1861-06-29 12",
    "1861-06-30 12", "1861-07-27 12", "1751-01-01", "1751-01-01",
    "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01",
    "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01",
    "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01",
    "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01",
    "1751-01-01", "1751-01-01", "1751-01-01", "1751-01-01", "1861-07-28 12",
    "1861-07-29 12", "1861-07-30 12", "1861-07-31 12", "1861-07-01 12",
    "1861-07-02 12", "1861-07-03 12", "1861-07-04 12", "1861-07-05 12",
...

I can't figure out directly which file are wrong what exactly the problem, but might be the file hd.1861-07.nc

This leads to crash of the CMOR tool:
```
^[[2;31;47mC Traceback:
! In function: cmor_check_monotonic
! called from: cmor_write_var_to_file
! called from: cmor_write
! ^[[0m

^[[1;31;47m!!!!!!!!!!!!!!!!!!!!!!!!!
!
! Error: axis time (table: Oday) has overlapping bounds values:
! 40361.000000, 40362.000000, 40331.000000 at index: 0
!
!!!!!!!!!!!!!!!!!!!!!!!!!^[[0m

forrtl: error (78): process killed (SIGTERM)
Image              PC                Routine
....
```

**Comment by tjiputra** [comment]
This was caused by two July files: 
- N1850_f19_tn14_20190730esm.micom.hd.1861-07.nc
- N1850_f19_tn14_20190730esm.micom.hd.1861-07b.nc

which was created when the simulation was archived in the middle of model execution.
These two files have been concatenated into one and the partial files have been removed.

**Comment by YanchunHe** [comment]
CMORization finished, and ready to publish to ESGF

**Versions**
- v20190920

**other information**
- PrePARE quality-check, passed, 5013 files scanned.
- sha256sum, calculated

**Comment by monsieuralok** [comment]
@YanchunHe Published

**Comment by YanchunHe** [comment]
`time_bnds` in some file(s) of daily ocean daily hbgcd output between  year 1861-1870 are wrong.

Please check if is that true and fix it.

It fails when processing the daily `chlos` and `phycos` filelds.


```
!
! Warning: axis time (table: Oday) has bounds values that leave gaps
! (index 0): 40330.000000, 40331.000000, 40357.000000
!
!!!!!!!!!!!!!!!!!!!!!!!!!^[[0m


^[[2;31;47mC Traceback:
! In function: cmor_check_monotonic
! called from: cmor_write_var_to_file
! called from: cmor_write
! ^[[0m

^[[1;31;47m!!!!!!!!!!!!!!!!!!!!!!!!!
!
! Error: axis time (table: Oday) has overlapping bounds values:
! 40361.000000, 40362.000000, 40331.000000 at index: 0
!
!!!!!!!!!!!!!!!!!!!!!!!!!^[[0m
```

@tjiputra  @JorgSchwinger

**Comment by YanchunHe** [comment]
Redo CMORization with additional fields. Ready to publish to ESGF

**data path**
- /tos-project1/NS9034K/CMIP6/CMIP/NCC/NorESM2-LM/esm-piControl

**version**
- v20191108

**sha256sum**
- .r1i1p1f1.sha256sum_v20191108b

**Comment by YanchunHe** [comment]
> `time_bnds` in some file(s) of daily ocean daily hbgcd output between year 1861-1870 are wrong.
> 
> Please check if is that true and fix it.
> 
> It fails when processing the daily `chlos` and `phycos` filelds.
> 
> ```
> !
> ! Warning: axis time (table: Oday) has bounds values that leave gaps
> ! (index 0): 40330.000000, 40331.000000, 40357.000000
> !
> !!!!!!!!!!!!!!!!!!!!!!!!!^[[0m
> 
> 
> ^[[2;31;47mC Traceback:
> ! In function: cmor_check_monotonic
> ! called from: cmor_write_var_to_file
> ! called from: cmor_write
> ! ^[[0m
> 
> ^[[1;31;47m!!!!!!!!!!!!!!!!!!!!!!!!!
> !
> ! Error: axis time (table: Oday) has overlapping bounds values:
> ! 40361.000000, 40362.000000, 40331.000000 at index: 0
> !
> !!!!!!!!!!!!!!!!!!!!!!!!!^[[0m
> ```
> 
> @tjiputra @JorgSchwinger

this was fixed in above cmorized data v20191108b (by tjiputra)

**Comment by monsieuralok** [comment]
@YanchunHe This dataset CMIP6.CMIP.NCC.NorESM2-LM.esm-piControl.r1i1p1f1.Ofx.sftof.gn exists in both previous and new version;  I guess it is not a problem.

**Comment by monsieuralok** [comment]
@YanchunHe published

**Comment by YanchunHe** [comment]
Cmorized with additional iLAMB variables (#262), AERday `zg500` (#263) and corrected `fNup` (#251).

They are ready to be published to ESGF.

**data path**
- /projects/NS9034K/CMIP6/.cmorout/NorESM2-LM/esm-piControl/v20210203
- /projects/NS9034K/CMIP6/CMIP/NCC/NorESM2-LM/esm-piControl

**version**
- v20210203

**sha256sum**
/projects/NS9034K/CMIP6/CMIP/NCC/NorESM2-LM/esm-piControl
- .r1i1p1f1.sha256sum_v20210203
(note, `zg500` and `fLitterSoil` are not available in the model output so not cmorized)

**Comment by monsieuralok** [comment]
@YanchunHe published

**Comment by YanchunHe** [comment]
New dataset version to fix issues #270 #273 is ready to be published:

**data path**
- /projects/NS9034K/CMIP6/.cmorout/NorESM2-LM/esm-piControl
- /projects/NS9034K/CMIP6/CMIP/NCC/NorESM2-LM/esm-piControl

**version**
- v20210319

**sha256sum**
/projects/NS9034K/CMIP6/CMIP/NCC/NorESM2-LM/esm-piControl
- .r1i1p1f1.sha256sum_v20210319

**Comment by monsieuralok** [comment]
@YanchunHe published
